manifests/provider/ad.pp

```diff
@@ -0,0 +1,135 @@
+  Optional[Array[String]]                                    $ad_gpo_map_deny                          = undef,
+  Optional[
+    Enum[
+      'interactive',
+      'remote_interactive',
+      'network',
+      'batch',
+      'service',
+      'permit',
+      'deny'
+      ]
+  ]                                                          $ad_gpo_default_right                     = undef,
+  Optional[Integer[0]]                                       $ad_maximum_machine_account_password_age  = undef,
+  Optional[Pattern['^\d+:\d+$']]                             $ad_machine_account_password_renewal_opts = undef,
+  Boolean                                                    $dyndns_update                            = true,
+  Optional[Integer]                                          $dyndns_ttl                               = undef,
+  Optional[Array[String]]                                    $dyndns_ifaces                            = undef,
+  Optional[Integer]                                          $dyndns_refresh_interval                  = undef,
+  Optional[Boolean]                                          $dyndns_update_ptr                        = undef,
+  Optional[Boolean]                                          $dyndns_force_tcp                         = undef,
+  Optional[Simplib::Hostname]                                $dyndns_server                            = undef,
+  Optional[String]                                           $override_homedir                         = undef,
+  Optional[Stdlib::Absolutepath]                             $homedir_substring                        = undef,
+  Optional[Boolean]                                          $krb5_use_enterprise_principal            = undef,
+  Optional[Variant[Enum['none'],Stdlib::Absolutepath]]       $krb5_confd_path                          = undef,
+  Boolean                                                    $ldap_id_mapping                          = true,
+  Optional[Integer[0]]                                       $ldap_idmap_range_min                     = undef,
+  Optional[Integer[1]]                                       $ldap_idmap_range_max                     = undef,
+  Optional[Integer[1]]                                       $ldap_idmap_range_size                    = undef,
+  Optional[String]                                           $ldap_idmap_default_domain_sid            = undef,
+  Optional[String]                                           $ldap_idmap_default_domain                = undef,
+  Optional[Boolean]                                          $ldap_idmap_autorid_compat                = undef,
+  Optional[Integer[1]]                                       $ldap_idmap_helper_table_size             = undef
+) {
+  include '::sssd'
+
+  concat::fragment { "sssd_${name}_ad_provider.domain":
+    target  => '/etc/sssd/sssd.conf',
+    content => template("${module_name}/provider/ad.erb")
```

Comment:
@jeefberkey It has the same tests as all of the rest of them.

Yes, we really should go through and do full O(n!) tests on the parameter permutations...but we currently don't.